pkg/agent/cniserver/server_windows.go

```diff
@@ -36,3 +36,9 @@ func updateResultDNSConfig(result *current.Result, cniConfig *CNIConfig) {
 	}
 	klog.Infof("Got runtime DNS configuration: %v", result.DNS)
 }
+
+// getInfraContainerID returns the K8S_POD_INFRA_CONTAINER_ID field of CNIConfig on Windows, and the containerID is the
+// actual container ID. kubelet-windows sends CNIAdd request for all containers in the same Pod.
+func (c *CNIConfig) getInfraContainerID() string {
```

Comment:
From kubelet code, container_id is always consistent with K8S_POD_INFRA_CONTAINER_ID.
And the behavior of windows should be:
1. kubelet calls CNIAdd with sandbox container id to set up network for this Pod.
2. kubelet calls CNIAdd with workload container id to get IP of this Pod.
This was a workaround for windows when sandbox is not in use: https://github.com/kubernetes/kubernetes/blob/961a5ed144cf0f2f4e2d858beb8cc0cfdd822c1b/pkg/kubelet/dockershim/docker_sandbox.go#L444.
So changing to  K8S_POD_INFRA_CONTAINER_ID doesn't prevent the race condition whatever it is.
My questions is, whether there's a real race condition happening? From the code, CNIAdd for setting up network and CNIAdd for getting IP are in different phase, so should not be called concurrently.
If we really need a lock to prevent potential race condition, it should be the K8S_POD_NAMESPACE+K8S_POD_NAME

Reply:
From my observation on the logs in cni server, the CNI Add request using infra cntainer is not a one-time request, so I think the infra container is also used in status query(not only the workload container). Besides, we might not only think of the "add" process, the race condition also exists when a cni query and delete happens in the same time. So using a unique key is a better choice.

Reply:
As for the infra_container_id, you are correct, it is the same as the container ID. Then I will change to use K8S_POD_NAMESPACE+K8S_POD_NAME as the container key